### PR TITLE
feat(desktop/communities): show community users which will be automatically joined after the ownership change in the All Members tab

### PR DESCRIPTION
**Pull request by mprakhov**
### What does the PR do

- show community users who will be automatically joined after the ownership change in the `All Members` tab
- show a specific icon for auto-rejoined members in `All members` tab
- refactored my requests to join and my awaiting requests to join logic

Closes: https://github.com/status-im/status-desktop/issues/12577 and https://github.com/status-im/status-desktop/issues/12776

status-go PR: https://github.com/status-im/status-go/pull/4409
### Affected areas

everything related to the community requests to join

### Screenshot of functionality

https://github.com/status-im/status-desktop/assets/117639195/ad46e55c-7937-4211-b369-112bccad950f

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Components/StatusMemberListItem.qml [hunk 1]
```diff
@@ -84,6 +84,12 @@ StatusListItem {
     */
     property bool isAdmin: false
 
+    /*!
+       \qmlproperty string StatusMemberListItem::isAwaitingAddress
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Components/StatusMemberListItem.qml:88]:
```suggestion
       \qmlproperty bool StatusMemberListItem::isAwaitingAddress
```

File: ui/StatusQ/src/StatusQ/Components/StatusMemberListItem.qml [hunk 2]
```diff
@@ -113,16 +119,9 @@ StatusListItem {
...
-        trustIndicator: {
-            if (root.isVerified)
-                return StatusContactVerificationIcons.TrustedType.Verified
-            if (root.isUntrustworthy)
-                return StatusContactVerificationIcons.TrustedType.Untrustworthy
-            return StatusContactVerificationIcons.TrustedType.None
-        } 
-    }
+    statusListItemTitleIcons.sourceComponent: if (root.isAwaitingAddress) return awaitingAddressComponent
+                                              else return statusContactVerificationIcons
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Components/StatusMemberListItem.qml:123]: No else after return pls

Comment (caybro) [reply ui/StatusQ/src/StatusQ/Components/StatusMemberListItem.qml:123]:
```suggestion
statusListItemTitleIcons.sourceComponent: root.isAwaitingAddress ? awaitingAddressComponent : statusContactVerificationIcons
```

**Review by igor-sirotin — APPROVED** [review]
I'm not familiar with this code, but it looks good to me 👍

**Review by MishkaRogachev — APPROVED** [review]
Great work!

**Pull request merged**